library/core/src/slice/mod.rs

```diff
@@ -5575,6 +5612,86 @@ unsafe impl<'a, T, const N: usize> TrustedRandomAccess for ArrayChunks<'a, T, N>
     }
 }
 
+/// A windowed iterator over a slice in overlapping chunks (`N` elements at a
+/// time), starting at the beginning of the slice
+///
+/// This struct is created by the [`array_windows`] method on [slices].
+///
+/// [`array_windows`]: ../../std/primitive.slice.html#method.array_windows
+#[derive(Debug, Clone, Copy)]
+#[unstable(feature = "array_windows", issue = "74985")]
+pub struct ArrayWindows<'a, T: 'a, const N: usize> {
+    slice_head_ptr: *const T,
+    num: usize,
+    marker: marker::PhantomData<&'a [T; N]>,
+}
+
+#[unstable(feature = "array_windows", issue = "74985")]
+impl<'a, T, const N: usize> Iterator for ArrayWindows<'a, T, N> {
+    type Item = &'a [T; N];
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.num == 0 {
+            return None;
+        }
+        // This is safe because it's indexing into a slice guaranteed to be length > N.
+        let ret = unsafe { self.slice_head_ptr.cast::<[T; N]>().as_ref() }.unwrap();
+        self.slice_head_ptr = unsafe { self.slice_head_ptr.add(1) };
+
+        self.num -= 1;
+        Some(ret)
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (self.num, Some(self.num))
+    }
+
+    #[inline]
+    fn count(self) -> usize {
+        self.num
+    }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        if self.num < n || self.num == 0 {
```

Comment:
```suggestion
        if self.num <= n {
```

You currently allow `nth` to access oob.

For example if there are still two windows left, `n == 2` would be out of bounds